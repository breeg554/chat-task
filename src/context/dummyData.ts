import img from "../media/example.jpg";

export const dummyUsers = {
  current: {
    id: 1,
    name: "Krystian Pach",
    avatar: "https://userstock.io/data/wp-content/uploads/2020/06/tyler-nix-PQeoQdkU9jQ-1024x1024.jpg",
  },
  member: {
    id: 2,
    name: "John Doe",
    avatar:
      "https://userstock.io/data/wp-content/uploads/2020/06/women-s-white-and-black-button-up-collared-shirt-774909-2-1024x1024.jpg",
  },
};

export const dummyMessages = [
  {
    id: 1,
    createdAt: 1641932804208,

    text: "Hello how are you?!",
    author: {
      id: 1,
      name: "Krystian Pach",
      avatar: "https://userstock.io/data/wp-content/uploads/2020/06/tyler-nix-PQeoQdkU9jQ-1024x1024.jpg",
    },
  },
  {
    id: 2,
    createdAt: 1641932801208,
    text: "I am fine!",
    author: {
      id: 2,
      name: "John Doe",
      avatar:
        "https://userstock.io/data/wp-content/uploads/2020/06/women-s-white-and-black-button-up-collared-shirt-774909-2-1024x1024.jpg",
    },
  },
  {
    id: 3,
    createdAt: 1621932804208,
    text: "Ok!",
    author: {
      id: 1,
      name: "Krystian Pach",
      avatar: "https://userstock.io/data/wp-content/uploads/2020/06/tyler-nix-PQeoQdkU9jQ-1024x1024.jpg",
    },
  },
  {
    id: 4,
    createdAt: 1641932804108,
    text: "Hello world!",
    author: {
      id: 2,
      name: "John Doe",
      avatar:
        "https://userstock.io/data/wp-content/uploads/2020/06/women-s-white-and-black-button-up-collared-shirt-774909-2-1024x1024.jpg",
    },
  },
  {
    id: 5,
    createdAt: 1641932804208,
    text: "Hello!",
    author: {
      id: 1,
      name: "Krystian Pach",
      avatar: "https://userstock.io/data/wp-content/uploads/2020/06/tyler-nix-PQeoQdkU9jQ-1024x1024.jpg",
    },
  },
  {
    id: 6,
    createdAt: 1641932904208,
    updatedAt: 1641922804208,
    text: "Hi!",
    author: {
      id: 2,
      name: "John Doe",
      avatar:
        "https://userstock.io/data/wp-content/uploads/2020/06/women-s-white-and-black-button-up-collared-shirt-774909-2-1024x1024.jpg",
    },
  },
  {
    id: 7,
    createdAt: 1641931802208,
    text: "Hello!",
    author: {
      id: 1,
      name: "Krystian Pach",
      avatar: "https://userstock.io/data/wp-content/uploads/2020/06/tyler-nix-PQeoQdkU9jQ-1024x1024.jpg",
    },
  },
  {
    id: 77,
    createdAt: 1641981899208,
    files: [img],
    description: "Check this out!",
    author: {
      id: 1,
      name: "Krystian Pach",
      avatar: "https://userstock.io/data/wp-content/uploads/2020/06/tyler-nix-PQeoQdkU9jQ-1024x1024.jpg",
    },
  },
  {
    id: 8,
    createdAt: 1641932804208,
    text: "Hi!",
    author: {
      id: 2,
      name: "John Doe",
      avatar:
        "https://userstock.io/data/wp-content/uploads/2020/06/women-s-white-and-black-button-up-collared-shirt-774909-2-1024x1024.jpg",
    },
  },
];
