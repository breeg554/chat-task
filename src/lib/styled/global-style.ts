import { createGlobalStyle } from "styled-components";

export const GlobalStyle = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }
  html,input{
    font-family: 'Roboto', sans-serif;
    font-size: 10px;   //1rem = 10px
  }
  body,button,input{
    background-color: #fff;
    color: ${({ theme }) => theme.colors.core.primary}
  }

`;
