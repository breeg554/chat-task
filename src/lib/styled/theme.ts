export type ThemeType = typeof theme;

export const theme = {
  colors: {
    core: {
      primary: "#2C3E50",
      secondary: "#555555",
      tertiary: "rgba(0,0,0,0.25)",
      success: "#74DD88",
    },
    hover: {
      primary: "#222",
      secondary: "#444",
      tertiary: "rgba(0,0,0,0.4)",
      success: "#95ee99",
    },
  },
  gradients: {
    core: {
      primary: "linear-gradient(to right, #00c6ff, #0072ff)",
    },
    hover: {
      primary: "linear-gradient(to right, #00a5dd, #0061dd)",
    },
  },
};
