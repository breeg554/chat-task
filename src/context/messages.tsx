import React, { createContext, useContext, useState } from "react";
import { TextMessage, FileMessage } from "../types";
import { dummyMessages } from "./dummyData";

interface MessagesProviderProps {
  children: React.ReactNode;
}
interface Context {
  messages: (TextMessage | FileMessage)[];
  sendMessage: (newMsg: TextMessage | FileMessage) => void;
  updateMessage: (updated: TextMessage | FileMessage) => void;
  removeMessage: (id: number) => void;
}

const MessagesContext = createContext<Context>(null!);
export const useMessages = () => useContext(MessagesContext);

const MessagesProvider = ({ children }: MessagesProviderProps) => {
  const [messages, setMessages] = useState<(TextMessage | FileMessage)[]>([
    ...dummyMessages,
  ]);

  const sendMessage = (newMsg: TextMessage | FileMessage) => {
    setMessages([...messages, newMsg]);
  };
  const removeMessage = (id: number) => {
    setMessages([...messages].filter((msg) => msg.id !== id));
  };
  const updateMessage = (updated: TextMessage | FileMessage) => {
    setMessages([...messages].map((msg) => (msg.id === updated.id ? updated : msg)));
  };

  return (
    <MessagesContext.Provider
      value={{ messages, sendMessage, removeMessage, updateMessage }}
    >
      {children}
    </MessagesContext.Provider>
  );
};

export default MessagesProvider;
