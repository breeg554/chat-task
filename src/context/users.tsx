import React, { createContext, useContext, useState } from "react";
import { User } from "../types";
import { dummyUsers } from "./dummyData";

interface UsersProviderProps {
  children: React.ReactNode;
}

interface Context {
  users: { current: User; member: User };
  toggleUsers: () => void;
}
const UsersContext = createContext<Context>(null!);

export const useUsers = () => useContext(UsersContext);

const UsersProvider = ({ children }: UsersProviderProps) => {
  const [users, setUsers] = useState({ ...dummyUsers });

  const toggleUsers = () => {
    setUsers({ current: { ...users.member }, member: { ...users.current } });
  };
  return (
    <UsersContext.Provider value={{ users, toggleUsers }}>
      {children}
    </UsersContext.Provider>
  );
};

export default UsersProvider;
