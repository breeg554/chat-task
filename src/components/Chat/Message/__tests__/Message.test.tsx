import { render } from "@testing-library/react";
import { dummyMessages } from "../../../../context/dummyData";
import { Message } from "..";
import MessagesProvider from "../../../../context/messages";
import UsersProvider from "../../../../context/users";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../../lib/styled";
describe("Message component", () => {
  it("renders out and match the snapshot", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <UsersProvider>
          <MessagesProvider>
            <Message data={dummyMessages[0]} isOwner />
          </MessagesProvider>
        </UsersProvider>
      </ThemeProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
