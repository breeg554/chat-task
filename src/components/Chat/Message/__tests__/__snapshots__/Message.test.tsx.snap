// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component renders out and match the snapshot 1`] = `
<div>
  <article
    class="sc-dlVxhl krVXtK"
  >
    <div
      class="avatar-wrapper"
    >
      <img
        alt="Krystian Pach"
        class="sc-ieecCq fZbkdH"
        src="https://userstock.io/data/wp-content/uploads/2020/06/tyler-nix-PQeoQdkU9jQ-1024x1024.jpg"
      />
    </div>
    <div
      class="content-wrapper"
    >
      <div
        class="sc-kfPuZi hMSVLP"
      >
        <h2
          class="sc-eCImPb rPzDq"
        >
          Krystian Pach
        </h2>
        <p
          class="sc-hKwDye ebVDgo"
        >
          Hello how are you?!
        </p>
      </div>
      <footer
        class="sc-iJKOTD lgfMDf"
      >
        <span>
          a day ago
        </span>
        <button
          class="sc-llYSUQ iYtcME"
        >
          edit
        </button>
        <button
          class="sc-llYSUQ iYtcME"
        >
          delete
        </button>
      </footer>
    </div>
  </article>
</div>
`;
